src/ow1/heroes/genji.opy

```diff
@@ -18,9 +18,30 @@ rule "[genji.opy]: Initialize Genji":
 rule "[genji.opy]: Change dash damage":
     @Event eachPlayer
     @Hero genji
-    @Condition eventPlayer.isUsingAbility1() == true
+    @Condition eventPlayer.isUsingAbility1() == true and not eventPlayer.isUsingUltimate() == true
 
     eventPlayer.setDamageDealt(100)
     waitUntil(eventPlayer.isUsingAbility1() == false, 9999)
     eventPlayer.setDamageDealt(percent(OW1_GENJI_SHURIKEN_DAMAGE/OW2_GENJI_SHURIKEN_DAMAGE))    
 
+rule "[genji.opy]: Change ult damage":
+    @Event eachPlayer
+    @Hero genji
+    @Condition eventPlayer.isUsingUltimate() == true
+
+    eventPlayer.setDamageDealt(100)
+    waitUntil(eventPlayer.isUsingUltimate() == false, 9999)
+    if eventPlayer.isUsingAbility2() == true:
+        waitUntil(eventPlayer.isUsingAbility2() == false, 9999)
+        eventPlayer.setDamageDealt(percent(OW1_GENJI_SHURIKEN_DAMAGE/OW2_GENJI_SHURIKEN_DAMAGE))
+    else:
```

Comment:
Do we need this block of code?

Reply:
Yes, if you deflect right before his ult ends, it would cause it to be using the extra damage modifier. This prevents it so if his ult ends and you are still deflecting, you wont have the extra damage until deflect is over.

Reply:
Can you please comment that in the source code for future devs?

Reply:
Other than that, it looks good to merge

Reply:
Yes, done.

Reply:
Wait your comment still doesn't make sense to me. How do you get higher damage dealt % if no code can give you higher damage when you're deflecting or ulting?

Reply:
Like one abuse I can think of is to deflect immediately after ulting so your damage dealt resets back to 100+%, but that's not exactly the situation you're describing.